multiuser/api/che-multiuser-api-resource/src/main/java/org/eclipse/che/multiuser/resource/api/workspace/LimitsCheckingWorkspaceManager.java

```diff
@@ -193,11 +216,37 @@ void checkMaxEnvironmentRam(WorkspaceConfig config) throws ServerException {
 
   @VisibleForTesting
   void checkRamResourcesAvailability(
-      String accountId, String namespace, WorkspaceConfig config, @Nullable String envName)
+      String accountId, String namespace, Workspace workspace, @Nullable String envName)
+      throws NotFoundException, ServerException, ConflictException {
+    if (workspace.getConfig() != null) {
+      checkRamResourcesAvailability(
+          accountId, namespace, workspace.getId(), workspace.getConfig(), envName);
+    } else {
+      LOG.warn(
+          "Can not estimate memory needed for devfile based workspace {}. Set memory resources limits may work incorrectly",
+          workspace.getId());
+    }
+  }
+
+  @VisibleForTesting
+  void checkRamResourcesAvailability(
+      String accountId,
+      String namespace,
+      String workspaceId,
+      WorkspaceConfig config,
+      @Nullable String envName)
       throws NotFoundException, ServerException, ConflictException {
     if (config.getEnvironments().isEmpty()) {
       return;
     }
+
+    if (SidecarToolingWorkspaceUtil.isSidecarBasedWorkspace(config.getAttributes())) {
+      LOG.warn(
+          "Sidecar memory of plugins of workspace `{}` is not taken in account. "
```

Comment:
How about: ``"The memory requirements for the plugin sidecars of workspace `{}` are not taken into account. The memory limits may not be applied correctly."``